File: receptionHubBackend/Controllers/Dtos/GostDto.cs
using receptionHubBackend.Models.Enums;

namespace receptionHubBackend.Controllers.Dtos;

public class GostDto
{
    public int IdGosta { get; set; }
    public TipGosta TipGosta { get; set; }
    public string? Ime { get; set; }
    public string? Prezime { get; set; }
    public string? NazivFirme { get; set; }
    public string? KontaktOsoba { get; set; }
    public string? BrojTelefona { get; set; }
    public string? Email { get; set; }
    public string? Drzava { get; set; }
    public string TipDokumenta { get; set; } = null!;
    public string? SlikaDokumenta { get; set; }
    public bool VIPGost { get; set; }
    public string? Dodatno { get; set; }
    public DateTime VrijemeKreiranja { get; set; }
    public int RecepcionerId { get; set; }
    public int BrojRezervacija { get; set; }
}

File: receptionHubFrontend/src/types/dtos/gost.dto.ts
export enum TipGosta {
  FizickoLice = 0,
  PravnoLice = 1
}

export interface GostDto {
  idGosta: number;
  tipGosta: TipGosta;
  ime?: string;
  prezime?: string;
  nazivFirme?: string;
  kontaktOsoba?: string;
  brojTelefona?: string;
  email?: string;
  drzava?: string;
  tipDokumenta: string;
  slikaDokumenta?: string;
  vipGost: boolean;
  dodatno?: string;
  vrijemeKreiranja: string; // ISO datetime string
  recepcionerId: number;
  brojRezervacija: number;
}

export interface KreirajGostaDto {
  tipGosta: TipGosta;
  ime?: string;
  prezime?: string;
  nazivFirme?: string;
  kontaktOsoba?: string;
  brojTelefona?: string;
  email?: string;
  drzava?: string;
  tipDokumenta: string;
  slikaDokumenta: string;
  vipGost?: boolean;
  dodatno?: string;
}

export interface AzurirajGostaDto {
  ime?: string;
  prezime?: string;
  nazivFirme?: string;
  kontaktOsoba?: string;
  brojTelefona?: string;
  email?: string;
  drzava?: string;
  tipDokumenta?: string;
  slikaDokumenta?: string;
  vipGost?: boolean;
  dodatno?: string;
}